client/src/app.js

```diff
@@ -125,45 +129,9 @@ export function checkStreamResult(result) {
             });
         };
 
-        rtcConn.onconnectionstatechange = function() {
-            if(rtcConn.connectionState === 'connected') { 
-                updateState({ established: true }); 
-            }
+        rtcConn.onconnectionstatechange = function() { handleConnectionStateChange(rtcConn); }
 
-            if(rtcConn.connectionState == 'disconnected' || 
-            rtcConn.connectionState == 'failed') { 
-                updateState({ 
-                    established: false,
-                    isPlaying: false
-                });
-            }
-
-        }
-
-        rtcConn.ondatachannel = (ev) => {
```

Comment:
Let's rename ev to dataChannelEvent and bring the function back here instead of in setMediaDescription.